### PR TITLE
Add syntax highlighting to the artifact text viewer

### LINKED ISSUE #2406: [FR] Syntax highlighting for text artifact viewer
I'm working on a syntax highlighting feature for the text artifact viewer.

highlight-demo

I used this package: https://github.com/conorhastings/react-syntax-highlighter

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Resolves #2406

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

The artifact text viewer supports syntax highlighting.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTextView.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { getSrc } from './ShowArtifactPage';
+import { Prism as SyntaxHighlighter } from 'react-syntax-highlighter';
+import { coy as style } from 'react-syntax-highlighter/dist/esm/styles/prism';
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTextView.js:5]: This page compares the themes `react-syntax-highlighter` supports:

https://k8shiro.github.io/ReactCompareCodeHighlighter/

**Comment by harupy** [comment]
Here's how it looks like:

Screen Shot 2020-07-04 at 2 41 11

**Review by smurching — COMMENTED** [review]
@harupy looks great! Would you mind adding a test case to `ShowArtifactTestView.test.js` that verifies that we render a `SyntaxHighlighter` with an appropriate `language` prop e.g. when rendering a `.py` artifact and a `MLproject` & `MLmodel` file? Would be nice to have a test that verifies the new behavior. Otherwise LGTM!

File: mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTextView.test.js [hunk 2]
```diff
@@ -55,11 +55,55 @@ describe('ShowArtifactTextView', () => {
...
+      done();
+    });
+  });
+
+  test('should render a python script without crashing', (done) => {
+    const getArtifact = jest.fn((artifactLocation) => {
+      return Promise.resolve('print("foo")');
+    });
+    const props = { path: 'fake.py', runUuid: 'fakeUuid', getArtifact: getArtifact };
+    wrapper = mount(<ShowArtifactTextView {...props} />);
```
Comment (smurching) [inline mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTextView.test.js:71]: I think if we use `shallow` to render the `ShowArtifactTextView`, we should actually be able to find a `SyntaxHighlighter` instance and assert on its props, e.g:

```
wrapper = shallow(<ShowArtifactTextView {...props} />);
setImmediate(() => {
  wrapper.update()
  expect(wrapper.find(SyntaxHighlighter).first().props().language).to.equal('py');
  done();
});
```

and same for the other tests (e.g. we can verify that MLproject and MLmodel files have yaml highlighting applied)

Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTextView.test.js]: Thanks! Fixed the tests.

**Comment by harupy** [comment]
I'll add a test for `getLanguage`.

File: mlflow/server/js/src/common/utils/FileUtils.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { getLanguage } from './FileUtils';
```
Comment (harupy) [inline mlflow/server/js/src/common/utils/FileUtils.test.js:1]: It turns out `getExtension` and `getBasename` in `FileUtils.js` are not tested.

Comment (harupy) [reply mlflow/server/js/src/common/utils/FileUtils.test.js:1]: I'll submit a follow-up PR to add tests for them.

**Review by smurching — APPROVED** [review]
LGTM thanks @harupy!

**Comment by smurching** [comment]
@harupy would you mind merging master to include your linter fix in this branch? Let's go ahead & merge after that

**Comment by harupy** [comment]
@smurching Merged the master branch

**Pull request merged**